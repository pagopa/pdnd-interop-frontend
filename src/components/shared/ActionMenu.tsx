import React, { useId } from 'react'
import { ClickAwayListener, MenuItem, Menu, IconButton, Skeleton, Box } from '@mui/material'
import type { ActionItem, ExtendedMUIColor } from '@/types/common.types'
import MoreVertIcon from '@mui/icons-material/MoreVert'
import { useTranslation } from 'react-i18next'

type ActionMenuProps = {
  actions: Array<ActionItem>
  iconColor?: ExtendedMUIColor
}

export const ActionMenu: React.FC<ActionMenuProps> = ({ actions, iconColor = 'primary' }) => {
  const [open, setOpen] = React.useState(false)
  const anchorRef = React.useRef<HTMLButtonElement>(null)
  const compositionButtonId = useId() + '-composition-button'
  const compositionMenuId = useId() + '-composition-menu'
  const { t } = useTranslation('shared-components', { keyPrefix: 'actionMenu' })

  const handleToggle = () => {
    setOpen((prevOpen) => !prevOpen)
  }

  const handleClose = (event: Event | React.SyntheticEvent) => {
    if (anchorRef.current && anchorRef.current.contains(event.target as HTMLElement)) {
      return
    }

    setOpen(false)
  }

  function handleListKeyDown(event: React.KeyboardEvent) {
    if (event.key === 'Tab') {
      event.preventDefault()
      setOpen(false)
    } else if (event.key === 'Escape') {
      setOpen(false)
    }
  }

  // return focus to the button when we transitioned from !open -> open
  const prevOpen = React.useRef(open)
  React.useEffect(() => {
    if (prevOpen.current === true && open === false) {
      anchorRef.current!.focus()
    }

    prevOpen.current = open
  }, [open])

  return (
    <>
      <IconButton
        ref={anchorRef}
        id={compositionButtonId}
        sx={{ visibility: actions.length > 0 ? 'visible' : 'hidden' }}
        aria-controls={open ? compositionMenuId : undefined}
        aria-expanded={open ? 'true' : undefined}
        aria-haspopup="true"
        onClick={handleToggle}
        aria-label={t('iconButtonAriaLabel')}
      >
        <MoreVertIcon color={iconColor} fontSize="small" />
      </IconButton>
      <ClickAwayListener onClickAway={handleClose}>
        <Menu
          open={open}
          id={compositionMenuId}
          anchorEl={anchorRef.current}
          aria-labelledby={compositionButtonId}
          onClose={handleClose}
          onKeyDown={handleListKeyDown}
          MenuListProps={{
            'aria-labelledby': 'basic-button',
          }}
          anchorOrigin={{ vertical: 'bottom', horizontal: 'right' }}
          transformOrigin={{ vertical: 'top', horizontal: 'right' }}
        >
          {actions.map(({ label, action }) => (
            <MenuItem
              key={label}
              onClick={(e) => {
                action()
                handleClose(e)
              }}
            >
              {label}
            </MenuItem>
          ))}
        </Menu>
      </ClickAwayListener>
    </>
  )
}

export const ActionMenuSkeleton: React.FC = () => {
  return (
    <Box component="span" sx={{ ml: 2, display: 'inline-block' }}>
      <Skeleton variant="rectangular" sx={{ my: 1, mx: 2 }} width={4} />
    </Box>
  )
}
