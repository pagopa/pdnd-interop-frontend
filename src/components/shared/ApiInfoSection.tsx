import React from 'react'
import { Trans, useTranslation } from 'react-i18next'
import { SectionContainer, SectionContainerSkeleton } from '../layout/containers'
import { type SkeletonProps, Stack, Typography } from '@mui/material'
import { RouterLink } from '@/router'
import { InformationContainer } from '@pagopa/interop-fe-commons'

type ApiInfoSectionProps = {
  ids: Array<{ name: string; id: string }>
}

export const ApiInfoSection: React.FC<ApiInfoSectionProps> = ({ ids }) => {
  const { t } = useTranslation('shared-components', {
    keyPrefix: 'apiInfoSection',
  })

  return (
    <SectionContainer
      title={t('title')}
      description={
        <Trans components={{ 1: <RouterLink to="SUBSCRIBE_INTEROP_M2M" /> }}>{t('content')}</Trans>
      }
    >
      {ids.map((element, index) => (
        <InformationContainer
          key={index}
          content={element.id}
          copyToClipboard={{ value: element.id, tooltipTitle: t('tooltipTitle') }}
          direction="column"
          label={element.name}
        />
      ))}
    </SectionContainer>
  )
}

export const ApiInfoSectionSkeleton: React.FC<SkeletonProps> = (props) => {
  return <SectionContainerSkeleton variant="rectangular" height={500} {...props} />
}
