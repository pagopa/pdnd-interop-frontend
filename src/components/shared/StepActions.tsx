import React from 'react'
import { Stack, Button } from '@mui/material'
import { RouteKey } from '@/router/router.types'
import { RouterLink } from '@/router'

type ActionButton = {
  label: string
  type: 'button'
  onClick: VoidFunction
}

type ActionLink = {
  label: string
  type: 'link'
  to: RouteKey
  disabled?: boolean
}

type ActionSubmit = {
  label: string
  type: 'submit'
  disabled?: boolean
}

export type BackAction = ActionButton | ActionLink
export type ForwardAction = ActionButton | ActionSubmit

type StepActionsProps = {
  back?: BackAction
  forward?: ForwardAction
}

export function StepActions({ back, forward }: StepActionsProps) {
  const forwardProps =
    forward &&
    (forward.type === 'button'
      ? { onClick: forward.onClick }
      : { type: 'submit', disabled: forward?.disabled })
  const backProps =
    back && back.type === 'link'
      ? { component: RouterLink, to: back.to }
      : { onClick: back?.onClick }

  return (
    <Stack direction="row" spacing={2} sx={{ mt: 2 }}>
      {back && (
        <Button variant="outlined" {...backProps}>
          {back.label}
        </Button>
      )}

      {forward && (
        <Button
          variant="contained"
          {...forwardProps}
          type={forwardProps?.type as 'submit' | 'button'}
        >
          {forward.label}
        </Button>
      )}
    </Stack>
  )
}
