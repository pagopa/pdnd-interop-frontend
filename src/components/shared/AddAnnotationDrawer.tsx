import type { RiskAnalysisTemplateAnswerAnnotation } from '@/api/purposeTemplate/mockedResponses'
import { Drawer } from '@/components/shared/Drawer'
import { RHFTextField } from '@/components/shared/react-hook-form-inputs'
import { Stack, Typography } from '@mui/material'
import React from 'react'
import { FormProvider, useForm } from 'react-hook-form'
import { useTranslation } from 'react-i18next'

type AddAnnotationFormValues = {
  annotation: RiskAnalysisTemplateAnswerAnnotation
}

type AddAnnotationDrawerProps = {
  isOpen: boolean
  onClose: VoidFunction
  onSubmit: (annotation: RiskAnalysisTemplateAnswerAnnotation) => void
}

export const AddAnnotationDrawer: React.FC<AddAnnotationDrawerProps> = ({
  isOpen,
  onClose,
  onSubmit,
}) => {
  const { t } = useTranslation('purposeTemplate', { keyPrefix: 'edit.step3.drawer' })
  const { t: tCommon } = useTranslation('common')

  const handleCloseDrawer = () => {
    onClose()
  }

  const formMethods = useForm<AddAnnotationFormValues>({
    defaultValues: {
      annotation: undefined,
    },
  })

  const _onSubmit = ({ annotation }: AddAnnotationFormValues) => {
    onClose()
    onSubmit(annotation)
  }

  const handleTransitionExited = () => {
    formMethods.reset()
  }

  const subtitle = (
    <Stack spacing={3} sx={{ mt: 2, mb: 0 }}>
      <Typography>{t('description')}</Typography>
      <Typography sx={{ fontWeight: 600 }}>{t('question')}</Typography>
    </Stack>
  )

  return (
    <FormProvider {...formMethods}>
      <Drawer
        isOpen={isOpen}
        onClose={handleCloseDrawer}
        title={t('title')}
        subtitle={subtitle}
        buttonAction={{
          label: tCommon('addBtn'),
          action: formMethods.handleSubmit(_onSubmit),
        }}
        onTransitionExited={handleTransitionExited}
      >
        <RHFTextField
          focusOnMount
          label={t('label')}
          size="medium"
          name="annotation"
          multiline
          rows={11}
          sx={{ mt: 0 }}
        />
      </Drawer>
    </FormProvider>
  )
}
