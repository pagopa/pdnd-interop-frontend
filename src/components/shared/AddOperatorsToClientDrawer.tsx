import type { RelationshipInfo } from '@/api/api.generatedTypes'
import { AuthHooks } from '@/api/auth'
import { PartyQueries } from '@/api/party'
import { Drawer } from '@/components/shared/Drawer'
import { RHFAutocompleteMultiple } from '@/components/shared/react-hook-form-inputs'
import { Alert, Stack } from '@mui/material'
import React from 'react'
import { FormProvider, useForm } from 'react-hook-form'
import { useTranslation } from 'react-i18next'

type AddSecurityOperatorFormValues = {
  selectedOperators: Array<RelationshipInfo>
}

type AddOperatorsDrawerProps = {
  isOpen: boolean
  onClose: VoidFunction
  excludeOperatorsIdsList: Array<string>
  onSubmit: (relationshipIds: Array<RelationshipInfo>) => void
}

export const AddOperatorsToClientDrawer: React.FC<AddOperatorsDrawerProps> = ({
  isOpen,
  onClose,
  excludeOperatorsIdsList,
  onSubmit,
}) => {
  const { t } = useTranslation('client', { keyPrefix: 'create.addOperatorsDrawer' })
  const { t: tCommon } = useTranslation('common')

  const handleCloseDrawer = () => {
    onClose()
  }

  const { jwt } = AuthHooks.useJwt()
  const formMethods = useForm<AddSecurityOperatorFormValues>({
    defaultValues: {
      selectedOperators: [],
    },
  })

  const { data: allPartyOperators = [], isLoading: isLoadingAllPartyOperators } =
    PartyQueries.useGetPartyUsersList(
      {
        productRoles: ['admin', 'security'],
        states: ['ACTIVE'],
        tenantId: jwt?.organizationId as string,
      },
      { suspense: false }
    )

  const availableOperators = allPartyOperators.filter(
    (partyOperator) => !excludeOperatorsIdsList.includes(partyOperator.id)
  )

  const options = availableOperators.map((o) => ({
    label: `${o.name} ${o.familyName}`,
    value: o,
  }))

  const _onSubmit = ({ selectedOperators }: AddSecurityOperatorFormValues) => {
    onClose()
    onSubmit(selectedOperators)
  }

  const handleTransitionExited = () => {
    formMethods.reset()
  }

  return (
    <FormProvider {...formMethods}>
      <Drawer
        isOpen={isOpen}
        onClose={handleCloseDrawer}
        title={t('title')}
        buttonAction={{
          label: tCommon('addBtn'),
          action: formMethods.handleSubmit(_onSubmit),
        }}
        onTransitionExited={handleTransitionExited}
      >
        <Stack spacing={3}>
          <RHFAutocompleteMultiple
            focusOnMount
            label={t('autocompleteInput.label')}
            labelType="external"
            size="small"
            name="selectedOperators"
            options={options}
            loading={isLoadingAllPartyOperators}
          />

          <Alert severity="info">{t('adminAlert')}</Alert>
        </Stack>
      </Drawer>
    </FormProvider>
  )
}
