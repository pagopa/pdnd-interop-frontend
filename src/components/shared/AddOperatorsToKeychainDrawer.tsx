import type { Users } from '@/api/api.generatedTypes'
import { AuthHooks } from '@/api/auth'
import { TenantQueries } from '@/api/tenant'
import { Drawer } from '@/components/shared/Drawer'
import { RHFAutocompleteMultiple } from '@/components/shared/react-hook-form-inputs'
import { Alert, Stack } from '@mui/material'
import { useQuery } from '@tanstack/react-query'
import React from 'react'
import { FormProvider, useForm } from 'react-hook-form'
import { useTranslation } from 'react-i18next'

type AddOperatorsFormValues = {
  selectedOperators: Users
}

type AddOperatorsDrawerProps = {
  isOpen: boolean
  onClose: VoidFunction
  excludeOperatorsIdsList: Array<string>
  onSubmit: (relationshipIds: Users) => void
}

export const AddOperatorsToKeychainDrawer: React.FC<AddOperatorsDrawerProps> = ({
  isOpen,
  onClose,
  excludeOperatorsIdsList,
  onSubmit,
}) => {
  const { t } = useTranslation('keychain', { keyPrefix: 'create.addOperatorsDrawer' })
  const { t: tCommon } = useTranslation('common')

  const handleCloseDrawer = () => {
    onClose()
  }

  const { jwt } = AuthHooks.useJwt()
  const formMethods = useForm<AddOperatorsFormValues>({
    defaultValues: {
      selectedOperators: [],
    },
  })

  const { data: allPartyOperators = [], isPending: isLoadingAllPartyOperators } = useQuery(
    TenantQueries.getPartyUsersList({
      roles: ['admin', 'security'],
      tenantId: jwt?.organizationId as string,
    })
  )

  const availableOperators = allPartyOperators.filter(
    (partyOperator) => !excludeOperatorsIdsList.includes(partyOperator.userId)
  )

  const options = availableOperators.map((o) => ({
    label: `${o.name} ${o.familyName}`,
    value: o,
  }))

  const _onSubmit = ({ selectedOperators }: AddOperatorsFormValues) => {
    onClose()
    onSubmit(selectedOperators)
  }

  const handleTransitionExited = () => {
    formMethods.reset()
  }

  return (
    <FormProvider {...formMethods}>
      <Drawer
        isOpen={isOpen}
        onClose={handleCloseDrawer}
        title={t('title')}
        buttonAction={{
          label: tCommon('addBtn'),
          action: formMethods.handleSubmit(_onSubmit),
        }}
        onTransitionExited={handleTransitionExited}
      >
        <Stack spacing={3}>
          <RHFAutocompleteMultiple
            focusOnMount
            label={t('autocompleteInput.label')}
            labelType="external"
            size="small"
            name="selectedOperators"
            options={options}
            loading={isLoadingAllPartyOperators}
          />
          <Alert severity="info">{t('adminAlert')}</Alert>
        </Stack>
      </Drawer>
    </FormProvider>
  )
}
