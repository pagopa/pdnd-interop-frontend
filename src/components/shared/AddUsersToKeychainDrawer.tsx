import type { Users } from '@/api/api.generatedTypes'
import { AuthHooks } from '@/api/auth'
import { TenantQueries } from '@/api/tenant'
import { Drawer } from '@/components/shared/Drawer'
import { RHFAutocompleteMultiple } from '@/components/shared/react-hook-form-inputs'
import { Alert, Stack } from '@mui/material'
import { useQuery } from '@tanstack/react-query'
import React from 'react'
import { FormProvider, useForm } from 'react-hook-form'
import { useTranslation } from 'react-i18next'

type AddUsersFormValues = {
  selectedUsers: Users
}

type AddUsersToKeychainDrawerProps = {
  isOpen: boolean
  onClose: VoidFunction
  excludeUsersIdsList: Array<string>
  onSubmit: (relationshipIds: Users) => void
}

export const AddUsersToKeychainDrawer: React.FC<AddUsersToKeychainDrawerProps> = ({
  isOpen,
  onClose,
  excludeUsersIdsList,
  onSubmit,
}) => {
  const { t } = useTranslation('keychain', { keyPrefix: 'create.addUsersToKeychainDrawer' })
  const { t: tCommon } = useTranslation('common')

  const handleCloseDrawer = () => {
    onClose()
  }

  const { jwt } = AuthHooks.useJwt()
  const formMethods = useForm<AddUsersFormValues>({
    defaultValues: {
      selectedUsers: [],
    },
  })

  const { data: allPartyUsers = [], isPending: isLoadingAllPartyUsers } = useQuery(
    TenantQueries.getPartyUsersList({
      roles: ['admin', 'security'],
      tenantId: jwt?.organizationId as string,
    })
  )

  const availableUsers = allPartyUsers.filter(
    (partyOperator) => !excludeUsersIdsList.includes(partyOperator.userId)
  )

  const options = availableUsers.map((o) => ({
    label: `${o.name} ${o.familyName}`,
    value: o,
  }))

  const _onSubmit = ({ selectedUsers }: AddUsersFormValues) => {
    onClose()
    onSubmit(selectedUsers)
  }

  const handleTransitionExited = () => {
    formMethods.reset()
  }

  return (
    <FormProvider {...formMethods}>
      <Drawer
        isOpen={isOpen}
        onClose={handleCloseDrawer}
        title={t('title')}
        buttonAction={{
          label: tCommon('addBtn'),
          action: formMethods.handleSubmit(_onSubmit),
        }}
        onTransitionExited={handleTransitionExited}
      >
        <Stack spacing={3}>
          <RHFAutocompleteMultiple
            focusOnMount
            label={t('autocompleteInput.label')}
            labelType="external"
            size="small"
            name="selectedUsers"
            options={options}
            loading={isLoadingAllPartyUsers}
          />
          <Alert severity="info">{t('adminAlert')}</Alert>
        </Stack>
      </Drawer>
    </FormProvider>
  )
}
