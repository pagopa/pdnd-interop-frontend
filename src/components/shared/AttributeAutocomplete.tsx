import React from 'react'
import { AttributeQueries } from '@/api/attribute'
import { RHFAutocompleteSingle } from '@/components/shared/react-hook-form-inputs'
import type { AttributeKey } from '@/types/attribute.types'
import { Button, Stack } from '@mui/material'
import { FormProvider, useForm } from 'react-hook-form'
import { useTranslation } from 'react-i18next'
import type { AttributeKind, DescriptorAttribute } from '@/api/api.generatedTypes'
import { useAutocompleteTextInput } from '@pagopa/interop-fe-commons'
import { keepPreviousData, useQuery } from '@tanstack/react-query'

export type AttributeAutocompleteProps = {
  attributeKey: AttributeKey
  onAddAttribute: (attribute: DescriptorAttribute) => void
  alreadySelectedAttributeIds: string[]
  direction?: 'column' | 'row'
}

type AttributeAutocompleteFormValues = { attribute: null | DescriptorAttribute }

export const AttributeAutocomplete: React.FC<AttributeAutocompleteProps> = ({
  attributeKey,
  onAddAttribute,
  alreadySelectedAttributeIds,
  direction = 'row',
}) => {
  const { t } = useTranslation('attribute', { keyPrefix: 'group' })
  const [attributeSearchParam, setAttributeSearchParam] = useAutocompleteTextInput()

  const attributeAutocompleteFormMethods = useForm<AttributeAutocompleteFormValues>({
    defaultValues: { attribute: null },
  })

  const { watch: watchAttribute, handleSubmit } = attributeAutocompleteFormMethods
  const selectedAttribute = watchAttribute('attribute')
  const isSelected = !!selectedAttribute

  /**
   * TEMP: This is a workaround to avoid the "q" param in the query to be equal to the selected attribute name.
   */
  function getQ() {
    let result = attributeSearchParam

    if (selectedAttribute && attributeSearchParam === selectedAttribute.name) {
      result = ''
    }

    return result
  }

  const { data } = useQuery({
    ...AttributeQueries.getList({
      kinds: [attributeKey.toUpperCase() as AttributeKind],
      q: getQ(),
      offset: 0,
      limit: 50,
    }),
    placeholderData: keepPreviousData,
  })

  const handleAddAttributeToGroup = handleSubmit(({ attribute }) => {
    if (!attribute) return
    onAddAttribute(attribute)
  })

  const options = React.useMemo(() => {
    const attributes = data?.results ?? []
    return attributes
      .filter((att) => !alreadySelectedAttributeIds.includes(att.id))
      .map((att) => ({
        label: att.name,
        value: att,
      }))
  }, [data?.results, alreadySelectedAttributeIds])

  return (
    <FormProvider {...attributeAutocompleteFormMethods}>
      <Stack direction={direction} alignItems="center" spacing={1}>
        <RHFAutocompleteSingle
          label={t('autocompleteInput.label')}
          placeholder={t('autocompleteInput.placeholder')}
          onInputChange={(_, value) => setAttributeSearchParam(value)}
          sx={{ mb: 0, flex: 1 }}
          options={options}
          focusOnMount
          name="attribute"
        />
        <Button
          onClick={handleAddAttributeToGroup}
          fullWidth={direction === 'column'}
          disabled={!isSelected}
          type="button"
          variant="contained"
        >
          {t('addBtn')}
        </Button>
      </Stack>
    </FormProvider>
  )
}
