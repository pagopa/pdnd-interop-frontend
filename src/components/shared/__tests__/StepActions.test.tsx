import React from 'react'
import { render } from '@testing-library/react'
import { type BackAction, type ForwardAction, StepActions } from '@/components/shared/StepActions'
import { MemoryRouter } from 'react-router-dom'
import { vi } from 'vitest'

const actions: Record<string, BackAction | ForwardAction> = {
  backButton: { label: 'label', type: 'button', onClick: vi.fn() },
  backLink: { label: 'label', type: 'link', to: 'TOS' },
  backLinkDisabled: { label: 'label', type: 'link', to: 'TOS', disabled: true },
  forwardButton: { label: 'label', type: 'button', onClick: vi.fn() },
  forwardSubmitDisabled: { label: 'label', type: 'submit', disabled: true },
}

const WrappedStepActions = (props: Record<string, BackAction | ForwardAction>) => (
  <MemoryRouter>
    <StepActions {...props} />
  </MemoryRouter>
)

describe("Checks that StepActions snapshots didn't change", () => {
  it('renders StepActions with back button', () => {
    const stepActions = render(<WrappedStepActions back={actions.backButton} />)

    expect(stepActions.baseElement).toMatchSnapshot()
  })

  it('renders StepActions with back link ', () => {
    const stepActions = render(<WrappedStepActions back={actions.backLink} />)

    expect(stepActions.baseElement).toMatchSnapshot()
  })

  it('renders StepActions with back link disabled', () => {
    const stepActions = render(<WrappedStepActions back={actions.backLinkDisabled} />)

    expect(stepActions.baseElement).toMatchSnapshot()
  })

  it('renders StepActions with forward button', () => {
    const stepActions = render(<WrappedStepActions forward={actions.forwardButton} />)

    expect(stepActions.baseElement).toMatchSnapshot()
  })

  it('renders StepActions with forward submit disabled', () => {
    const stepActions = render(<WrappedStepActions forward={actions.forwardSubmitDisabled} />)

    expect(stepActions.baseElement).toMatchSnapshot()
  })

  it('renders StepActions with forward and back button disabled', () => {
    const stepActions = render(
      <WrappedStepActions
        back={actions.backButton as BackAction}
        forward={actions.forwardSubmitDisabled}
      />
    )

    expect(stepActions.baseElement).toMatchSnapshot()
  })
})
