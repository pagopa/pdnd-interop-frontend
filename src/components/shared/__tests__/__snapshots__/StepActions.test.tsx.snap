// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Checks that StepActions snapshots didn't change > renders StepActions with back button 1`] = `
<body>
  <div>
    <div
      class="css-9v4aab-MuiStack-root"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        label
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</body>
`;

exports[`Checks that StepActions snapshots didn't change > renders StepActions with back link  1`] = `
<body>
  <div>
    <div
      class="css-9v4aab-MuiStack-root"
    >
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-i4i1vz-MuiTypography-root-MuiLink-root-MuiButtonBase-root-MuiButton-root"
        href="/it/termini-di-servizio"
        tabindex="0"
      >
        label
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </a>
    </div>
  </div>
</body>
`;

exports[`Checks that StepActions snapshots didn't change > renders StepActions with back link disabled 1`] = `
<body>
  <div>
    <div
      class="css-9v4aab-MuiStack-root"
    >
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-i4i1vz-MuiTypography-root-MuiLink-root-MuiButtonBase-root-MuiButton-root"
        href="/it/termini-di-servizio"
        tabindex="0"
      >
        label
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </a>
    </div>
  </div>
</body>
`;

exports[`Checks that StepActions snapshots didn't change > renders StepActions with forward and back button disabled 1`] = `
<body>
  <div>
    <div
      class="css-9v4aab-MuiStack-root"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1rwt2y5-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        label
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
        disabled=""
        tabindex="-1"
        type="submit"
      >
        label
      </button>
    </div>
  </div>
</body>
`;

exports[`Checks that StepActions snapshots didn't change > renders StepActions with forward button 1`] = `
<body>
  <div>
    <div
      class="css-9v4aab-MuiStack-root"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        label
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</body>
`;

exports[`Checks that StepActions snapshots didn't change > renders StepActions with forward submit disabled 1`] = `
<body>
  <div>
    <div
      class="css-9v4aab-MuiStack-root"
    >
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-sghohy-MuiButtonBase-root-MuiButton-root"
        disabled=""
        tabindex="-1"
        type="submit"
      >
        label
      </button>
    </div>
  </div>
</body>
`;
